### ISSUE TITLE
all-in-one benchmark tool benchmark issue, the latency of large batch is too high

### ISSUE THREAD

**Issue by Fred-cell** [intel-analytics/BigDL#10463]
benchmark chatglm3-6b on Arc A770 with 1k input and 512 output, the performance is as below:
![image](https://github.com/intel-analytics/BigDL/assets/99886928/72d0312a-5651-46c6-86f0-576b0931f451)

**Comment by hkvision** [comment]
Have you add quantize kv cache when you test this?

**Comment by Fred-cell** [comment]
yes!

**Comment by Fred-cell** [comment]
折纸的过程看似简单，其实想要做好，还是需要一套很复杂的工艺。以折一支玫瑰花为例，我们可以将整个折纸过程分成三个阶段，即：创建栅格折痕，制作立体基座，完成花瓣修饰。首先是创建栅格折痕：这一步有点像我们折千纸鹤的第一步，即通过对称州依次对折，然后按照长和宽两个维度，依次进行多等分的均匀折叠；最终在两个方向上的折痕会交织成一套完整均匀的小方格拼接图案；这些小方格就组成了类似二维坐标系的参考系统，使得我们在该平面上，通过组合临近折痕的方式从二维小方格上折叠出三维的高台或凹陷，以便于接下来的几座制作过程。需要注意的是，在建立栅格折痕的过程中，可能会出现折叠不对成的情况，这种错误所带来的后果可能是很严重的，就像是蝴蝶效应，一开始只是毫厘之差，最后可能就是天壤之别。然后是制作立体基座：在这一步，我们需要基于栅格折痕折出对称的三维高台或凹陷。从对称性分析不难发现，玫瑰花会有四个周对称的三维高台和配套凹陷。所以，我们可以先折出四分之一的凹陷和高台图案，然后以这四分之一的部分作为摸板，再依次折出其余三个部分的重复图案。值得注意的是，高台的布局不仅要考虑长和宽这两个唯独上的规整衬度和对称分布，还需要同时保证高这个维度上的整齐。与第一阶段的注意事项类似，请处理好三个维度上的所有折角，确保它们符合计划中所要求的那种布局，以免出现三维折叠过程中的蝴蝶效应；为此，我们常常会在折叠第一个四分之一图案的过程中，与成品玫瑰花进行反复比较，以便在第一时间排除掉所有可能的错误。最后一个阶段是完成花瓣修饰。在这个阶段，我们往往强调一个重要名词，叫用心折叠。这里的用心已经不是字面上的认真这个意思，而是指通过我们对于大自然中玫瑰花外型的理解，借助自然的曲线去不断修正花瓣的形状，以期逼近现实中的玫瑰花瓣外形。请注意，在这个阶段的最后一步，我们需要通过拉扯已经弯折的四个花瓣，来调整玫瑰花中心的绽放程度。这个过程可能会伴随玫瑰花整体结构的崩塌，所以，一定要控制好调整的力道，以免出现不可逆的后果。最终，经过三个阶段的折叠，我们会得到一支栩栩如生的玫瑰花冠。如果条件允许，我们可以在一根拉直的铁丝上缠绕绿色纸条，并将玫瑰花冠插在铁丝的一段。这样，我们就得到了一支手工玫瑰花。总之，通过创建栅格折痕，制作立体基座，以及完成花瓣修饰，我们从二维的纸面上创作出了一支三维的花朵。这个过程虽然看似简单，但它确实我们人类借助想象力和常见素材而创作出的艺术品。问: 请基于以上描述，分析哪些步骤做错了很大可能会导致最终折叠失败？

**Comment by hkvision** [comment]
Updated data from fred: when batch=8, the latency is not reasonable compared with 4/16.

**Comment by zhentaocc** [comment]
seems there is a sudden performance degradation at batch 7-8 ?
![image](https://github.com/intel-analytics/BigDL/assets/90437536/a7f10a01-f56c-49f5-ad65-d5eba6d87ef6)

**Comment by hkvision** [comment]
We have reproduced this issue. Seems something wrong with batch 7/8, we are looking into this.

**Comment by hkvision** [comment]
> chatglm3-6b with batch 8 has the same problem

The model you report in this issue is already chatglm3-6b. We are fixing it, should be fixed very soon.

**Comment by leonardozcm** [comment]
hi, may you try with the latest ipex-llm(>=0325)?

**Comment by Fred-cell** [comment]
Yes, FP8 with batch 8 is fixed, but there is the same issue for INT4.
0,THUDM/chatglm3-6b,1223.31,**47.75**,0.0,1024-512,8,634-174,1,sym_int4,N/A,8.43,5.236328125,N/A

**Comment by leonardozcm** [comment]
> Yes, FP8 with batch 8 is fixed, but there is the same issue for INT4. 0,THUDM/chatglm3-6b,1223.31,**47.75**,0.0,1024-512,8,634-174,1,sym_int4,N/A,8.43,5.236328125,N/A

Sorry that I can not reproduce this: 

```
,model,1st token avg latency (ms),2+ avg latency (ms/token),encoder time (ms),input/output tokens,batch_size,actual input/output tokens,num_beams,low_bit,cpu_embedding,model loading time (s),peak mem (GB),streaming
0,THUDM/chatglm3-6b,1528.2,31.47,0.0,634-174,8,642-174,1,sym_int4,N/A,6.54,6.419921875,N/A

```

Add chatglm3-6b bs2~8:
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/zhaoc1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/zhaoc1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">


bs2 | 14.38
-- | --
bs4 | 17.74
bs6 | 25.22
bs8 | 31.47



</body>

</html>

I am using latest ipex-llm(both ipex-llm&bigdl-core-xe == 2.5.0b20240326)
